app/controllers/pensions/incomeFromPensions/TaxPaidOnLumpSumAmountController.scala

```diff
@@ -0,0 +1,106 @@
+
+
+  def amountForm: Form[BigDecimal] = AmountForm.amountForm(
+    emptyFieldKey = "incomeFromPensions.taxPaidOnLumpSumAmount.error.error.noEntry",
+    wrongFormatKey = "incomeFromPensions.taxPaidOnLumpSumAmount.error.error.incorrectFormat",
+    exceedsMaxAmountKey = "incomeFromPensions.taxPaidOnLumpSumAmount.error.error.overMaximum"
+  )
+
+  def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit request =>
+    inYearAction.notInYear(taxYear) {
+      pensionSessionService.getPensionsSessionDataResult(taxYear, request.user) {
+        case Some(data) =>
+          val preAmount = data.pensions.incomeFromPensions.statePensionLumpSum.flatMap(_.taxPaid)
+          if (data.pensions.incomeFromPensions.statePensionLumpSum.flatMap(_.taxPaidQuestion).contains(true)) {
+            data.pensions.incomeFromPensions.statePensionLumpSum.flatMap(_.taxPaid) match {
+              case Some(amount) => Future.successful(Ok(taxPaidOnLumpSumAmountView(amountForm.fill(amount), taxYear, preAmount)))
+              case None => Future.successful(Ok(taxPaidOnLumpSumAmountView(amountForm, taxYear, None)))
+            }
+          } else {
+            Future.successful(Redirect(TaxPaidOnStatePensionLumpSumController.show(taxYear)))
+          }
+        case _ =>
+          //TODO redirect to the income from pensions CYA page
+          Future.successful(Redirect(PensionsSummaryController.show(taxYear)))
+      }
+    }
+  }
+
+  def submit(taxYear: Int): Action[AnyContent] = authAction.async { implicit request =>
+    inYearAction.notInYear(taxYear) {
+      pensionSessionService.getPensionsSessionDataResult(taxYear, request.user) {
+        case Some(data) =>
+          amountForm.bindFromRequest.fold(
+            formWithErrors => {
+              val taxPaid = data.pensions.incomeFromPensions.statePensionLumpSum.flatMap(_.taxPaid)
+              Future.successful(BadRequest(taxPaidOnLumpSumAmountView(formWithErrors, taxYear, taxPaid)))
+            },
+            amount => {
+              if (data.pensions.incomeFromPensions.statePensionLumpSum.flatMap(_.taxPaidQuestion).contains(true)) {
+                val pensionsCYAModel: PensionsCYAModel = data.pensions
+                val viewModel: IncomeFromPensionsViewModel = pensionsCYAModel.incomeFromPensions
+                val statePensionLumpSumModel: Option[StateBenefitViewModel] = viewModel.statePensionLumpSum
+                val updatedCyaModel: PensionsCYAModel = {
+                  pensionsCYAModel.copy(incomeFromPensions = viewModel.copy(statePensionLumpSum = statePensionLumpSumModel.map(_.copy(taxPaid = Some(amount)))))
+                }
+
+                pensionSessionService.createOrUpdateSessionData(request.user,
+                  updatedCyaModel, taxYear, data.isPriorSubmission)(errorHandler.internalServerError()) {
+                  Redirect(PensionsSummaryController.show(taxYear))
```

Comment:
This should have a TODO
However shouldn't it navigate to **state-pension-lump-sum-tax** page which is available

There appears to be other pages that haven't been linked up (I know we have a redirect task but we normally do this part as we go along)
Can we link **state-pension** to **state-pension-amount** as that page is there (assume there is a TODO for that but even if not can do?)

**state-pension-lump-sum-amount** should link to **state-pension-lump-sum-tax**

Reply:
This one is on successful submit so I've added a TODO here.  If the **state-pension-lump-sum-tax Question** is false, there is a redirect to that question page below. I will link up the other pages too